code/modules/events/disease_outbreak.dm

```diff
@@ -271,25 +279,40 @@
 				/datum/symptom/fire,
 				/datum/symptom/flesh_death,
 				/datum/symptom/flesh_eating,
-				/datum/symptom/narcolepsy,
-				/datum/symptom/voice_change,
+				/datum/symptom/visionloss,
 			)
 
-	for(var/i in 1 to max_symptoms)
+	var/current_severity = 0
+
+	while(symptoms.len < max_symptoms)
 		var/datum/symptom/chosen_symptom = pick_n_take(possible_symptoms)
-		if(chosen_symptom)
-			var/datum/symptom/new_symptom = new chosen_symptom
-			symptoms += new_symptom
 
-	if(prob(50))
-		symptoms += new /datum/symptom/viralevolution
+		if(!chosen_symptom)
+			CRASH("Advanced disease could not pick a symptom!")
 
-	if(prob(50))
-		symptoms += new /datum/symptom/youth
+		//Checks if the chosen symptom is severe enough to meet requested severity. If not, pick a new symptom.
+		//If we've met requested severity already, we don't care and will keep the chosen symptom.
+		var/datum/symptom/new_symptom = new chosen_symptom
 
-	Refresh()
+		if((current_severity < requested_severity) && (new_symptom.severity < requested_severity))
+			continue
+
+		symptoms += new_symptom
 
-	name = "Sample #[rand(1, 9999)]"
+		//Applies the illness name based on the most severe symptom.
+		if(new_symptom.severity > current_severity)
+			name = "[new_symptom.illness]"
+			current_severity = new_symptom.severity
+
+	//Finalize the generated virus with EY on the end.
```

Comment:
What's `EY`? It's not abundantly clear with the code (I presume it has something to do with the youth datum, but it's just not helpful in its current state). There's plenty of space to write out what needs to be written out without using acronyms.

Reply:
Okay, I'll expand it.